wasmdome/src/main.rs

```diff
@@ -103,6 +139,127 @@ NKEYs are sensitive and should be treated as secrets.
         Err("Did not obtain valid arena credentials".into())
     }
 }
+
+fn check_schedule(nc: nats::Connection) -> Result<(), Box<dyn Error>> {
+    let res = nc.request_timeout(
+        "wasmdome.public.arena.schedule",
+        "",
+        std::time::Duration::from_millis(500),
+    );
+
+    if let Err(e) = res {
+        println!("Error requesting schedule from the lattice: {}", e);
+        return Ok(());
+    };
+
+    let data = &res.unwrap().data;
+    let matches: Vec<StoredMatch> = serde_json::from_slice(data)?;
+    let mut table = Table::new();
+    if matches.len() > 0 {
+        table.add_row(row![
+            "Match Id",
+            "Match Start",
+            "Board Size (H ⨯ W)",
+            "Max Actors",
+            "Max Turns",
+            "Actions per turn"
+        ]);
+        matches.iter().for_each(|m| {
+            table.add_row(row![
+                format!("{}", m.match_id),
+                format!("{}", m.entry.match_start),
+                format!("{} ⨯ {}", m.entry.board_height, m.entry.board_width),
+                format!("{}", m.entry.max_actors),
+                format!("{}", m.entry.max_turns),
+                format!("{}", m.aps_per_turn),
+            ]);
+        });
+    } else {
+        println!("No schedule results available.");
+        return Ok(());
+    };
+
+    table.printstd();
+    Ok(())
+}
+
+fn run_match(
+    nc: nats::Connection,
+    max_turns: u32,
+    board_height: u32,
+    board_width: u32,
+    aps_per_turn: u32,
+) -> Result<(), Box<dyn Error>> {
+    // publish game start command onto local lattice (through nc)
+    // subscribe to arena events topic and then display match conclusion event, or timeout with an error
+
+    let req = nc.request_timeout(
```

Comment:
Subscribe to the arena event subject before the request, otherwise there's a chance we'll miss quick-firing arena events between the request call and the match start.

Reply:
This has been resolved

Reply:
Still looks like you're subscribing to the match events rather than the arena event stream

Reply:
Ah so not `wasmdome.public.arena.events`?

Reply:
Wasmdome.arena.public.events is right. GitHub was showing me the old subject for some reason

Reply:
Mine did that too, I had to hard refresh to get the new code to show up